src/Functions/toStartOfInterval.cpp

```diff
@@ -248,9 +257,12 @@ class FunctionToStartOfInterval : public IFunction
         result_data.resize(size);
 
         Int64 scale_multiplier = DecimalUtils::scaleMultiplier<DateTime64>(scale);
+        UInt8 week_mode = enable_default_monday_first ? 1 : 0;
 
         for (size_t i = 0; i != size; ++i)
-            result_data[i] = static_cast<ResultFieldType>(ToStartOfInterval<unit>::execute(time_data[i], num_units, time_zone, scale_multiplier));
+        {
+            result_data[i] = static_cast<ResultFieldType>(ToStartOfInterval<unit>::execute(time_data[i], num_units, time_zone, scale_multiplier, week_mode));
```

Comment:
Functions `ToStartOfXYZ::execute()` in DateTimeTransforms.h do a call on `time_zone`. It would be slightly prettier and more consistent with other places in the code if `time_zone` could be the last argument. Yes, that means
- moving `scale_multiplier` and `week_mode` after `num_units`
- removing the default `=1` argument on `week_mode`. Functions that can't do anything with it should rather ignore it right away.